pkg/system/phase4_configuring.go

```diff
@@ -677,6 +689,106 @@ func (r *Reconciler) prepareGCPBackingStore() error {
+		r.IBMCloudCOSCreds.Namespace = ibmCOSCredDefaultNS
+		util.KubeCheck(r.IBMCloudCOSCreds)
+		if r.IBMCloudCOSCreds.UID == "" {
+			r.IBMCloudCOSCreds.Namespace = targetNS
+		}
+	}
+	if r.IBMCloudCOSCreds.UID == "" {
+		r.Logger.Errorf("Cloud credentials secret %q is not ready yet", r.IBMCloudCOSCreds.Name)
+		return fmt.Errorf("Cloud credentials secret %q is not ready yet", r.IBMCloudCOSCreds.Name)
+	}
+
+	if val, ok := r.IBMCloudCOSCreds.StringData["IBM_COS_Endpoint"]; ok {
+		// Use the endpoint provided in the secret
+		endpoint = val
+		r.Logger.Infof("Endpoint provided in secret: %q", endpoint)
+		if val, ok := r.IBMCloudCOSCreds.StringData["IBM_COS_Location"]; ok {
+			location = val
+			r.Logger.Infof("Location provided in secret: %q", location)
+		}
+	} else {
+		// Endpoint not provided in the secret, construct one based on the cluster's region
+		// https://cloud.ibm.com/docs/cloud-object-storage?topic=cloud-object-storage-endpoints#endpoints
+		region, err := util.GetIBMRegion()
+		if err != nil {
+			r.Logger.Errorf("Failed to get IBM Region. %q", err)
+			return fmt.Errorf("Failed to get IBM Region")
+		}
+		r.Logger.Infof("Constructing endpoint for region: %q", region)
+		// https://cloud.ibm.com/docs/cloud-object-storage?topic=cloud-object-storage-classes#classes-locationconstraint
+		endpoint = fmt.Sprintf(ibmEndpoint, region)
+		location = fmt.Sprintf(ibmLocation, region)
+	}
+
+	if _, err := url.Parse(endpoint); err != nil {
+		r.Logger.Errorf("Invalid formate URL %q", endpoint)
+		return fmt.Errorf("Invalid formate URL %q", endpoint)
+	}
+
+	r.Logger.Infof("IBM COS Endpoint: %s   LocationConstraint: %s", endpoint, location)
+
+	var accessKeyID string
+	if val, ok := r.IBMCloudCOSCreds.StringData["IBM_COS_ACCESS_KEY_ID"]; ok {
+		accessKeyID = val
+	} else {
+		r.Logger.Errorf("Missing IBM_COS_ACCESS_KEY_ID in the secret")
+		return fmt.Errorf("Missing IBM_COS_ACCESS_KEY_ID in the secret")
+	}
+
+	var secretAccessKey string
+	if val, ok := r.IBMCloudCOSCreds.StringData["IBM_COS_SECRET_ACCESS_KEY"]; ok {
+		secretAccessKey = val
+	} else {
+		r.Logger.Errorf("Missing IBM_COS_SECRET_ACCESS_KEY in the secret")
+		return fmt.Errorf("Missing IBM_COS_SECRET_ACCESS_KEY in the secret")
+	}
+
+	bucketName := r.generateBackingStoreTargetName()
+	r.Logger.Infof("IBM COS Bucket Name: %s", bucketName)
+
+	s3Config := &aws.Config{
+		S3ForcePathStyle: aws.Bool(true),
+		Endpoint:         aws.String(endpoint),
+		Credentials: credentials.NewStaticCredentials(
+			accessKeyID,
+			secretAccessKey,
+			"",
+		),
+		Region: &location,
+	}
+	if err := r.createS3BucketForBackingStore(s3Config, bucketName); err != nil {
+		return err
+	}
+	r.Logger.Infof("Creadted bucket: %s", bucketName)
+
+	// create backing store
+	r.DefaultBackingStore.Spec.Type = nbv1.StoreTypeIBMCos
+	r.DefaultBackingStore.Spec.IBMCos = &nbv1.IBMCosSpec{
+		TargetBucket: bucketName,
+		Secret: corev1.SecretReference{
+			Name:      r.IBMCloudCOSCreds.Name,
```

Comment:
Just use the name and namespace from above. No need for the reconcile object.

Reply:
@jeniawhite lets expand what you are saying here - should we detect the IBM cloud platform and then create the default backing store directly with the fixed secret name, without even trying to load it first? In case the secret does not exist this means our default backing store will already be created this way and will wait for the secret to appear.
On the other hand, if we do want the system reconciler to check the secret exists before deciding what is the default backing store - saw we want to decide to create a different backing store, or some other logic in the scope of the system reconciler.

I actually don't mind which option is preferred, but I would like to streamline with the rest of the cloud platform flows. So I would still write it just like we do for AWS creds and if we decide to refactor later we can do all platforms the same.